uswds/uswds
# USWDS - Dependencies: Remove `handlebars-helpers`

# Summary

Removes `handlebars-helpers` which is an unused dependency. 

## Breaking change

This is not a breaking change.

## Related issue

Closes https://github.com/uswds/uswds/security/dependabot/14

## Preview link

Storybook preview 

## Problem statement

`handlebars-helpers@0.10.0` requires `highlight.js@^9.12.0` via `helper-markdown@1.0.0`

`highlight.js` introduces a `Moderate` ReDOS vulnerability

## Solution

Remove `handlebars-helpers`.

USWDS no longer uses handlebars / mustache templating language. We do not utilize the package using its Usability Guidance provided.

Additionally, we removed this package in [[2.14.0](https://github.com/uswds/uswds/discussions/5223)](https://github.com/uswds/uswds/discussions/5223)

In 3.x, removed `handlebars` but not `handlebars-helpers` in [[3.0.0 beta 5](https://github.com/uswds/uswds/releases/tag/v3.0.0-beta.5)](https://github.com/uswds/uswds/releases/tag/v3.0.0-beta.5)

## Testing and review

1. Pull branch.
2. Do a clean install using `npm ci` and confirm there are no installation errors.
3. Run `npm run build` and confirm there are no build errors or unintended file changes.
4. Run `npm run build:storybook` to confirm there are no storybook compilation errors.
5. Run `npm run build:html` to confirm there are no compiled changes.
6. Run `npm run start` .
7. Confirm that there are no errors or visual regressions.

### Testing Checklist

- [ ]  No installation errors.
- [ ]  No storybook errors.
- [ ]  No site build errors.
- [ ]  No visual regression.
- [ ]  No unintended changed files.

## Dependency updates

| Dependency name | Previous version | New version |
| --- | --- | --- |
| handlebars-helpers | `0.10.0` | - |
